Command/GenerateRoutes.php

```diff
@@ -55,16 +75,32 @@ protected function execute(InputInterface $input, OutputInterface $output): int
                     unset($routeData[$key]);
                 }
             }
-            $routes[$name] = $routeData;
+
+            if ($this->filterMatches($name, $routeData, $namePrefix)) {
+                $routes[$name] = $routeData;
+            }
         }
 
         $targetPath = $this->projectDirectory . '/fos-routing.yaml';
-        $output->writleln(Yaml::dump($routes));
+        $output->writeln(Yaml::dump($routes));
 
         $io = new SymfonyStyle($input, $output);
 
         $io->info('If you do not like yaml. You can use the symplify/config-transformer package to change it into php.');
 
         return Command::SUCCESS;
     }
+
+    private function filterMatches(string $name, array $data, ?string $namePrefix, ?string $controllerName): bool
+    {
+        if ($namePrefix !== null) {
+            return str_starts_with($name, $namePrefix);
+        }
+
+        if ($controllerName !== null) {
+            return $data['controller'] !== $controllerName;
```

Comment:
also `str_starts_with` as previousy the controller service included more actions

Reply:
I mean you can't use both filters in combination but that doesn't sound like a use-case.